docs/sbom.md

```diff
@@ -0,0 +1,417 @@
+
+## Types
+
+Depending on how an SBOM is generated, it will contain varying levels of information. The sections below highlight
+different viewpoints of when an SBOM is created and what type of data it includes.
+
+### Build vs Release
+
+_Build-time_ SBOMs are created during the initial build process of an artifact (for example, when an RPM is created
+from source files or a container image is built using [`buildah`](https://buildah.io/)). These SBOMs document the
+components used during the build process to produce the final artifact as well as any components used for the build
+process itself. This SBOM type also aligns with the _Build_ SBOM type from CISA's guidance on
+[Types of SBOM Documents](https://www.cisa.gov/sites/default/files/2023-04/sbom-types-document-508c.pdf).
+
+_Release-time_ SBOMs are generated when an artifact is released or published. These SBOMs build upon build-time
+SBOMs by incorporating additional metadata, such as the repositories or locations where the artifact is
+published, and associating it with the relevant product information if there is any. Release-time SBOMs reflect the
+state of the software as it is distributed to end users. This SBOM type is close to the _Deployed_ type as defined
+by CISA, but it reflects the state of the product that _would_ be installed by a given end user.
+
+Red Hat's publicly available SBOMs are of the "release" type, including details about where an artifact
+can be located after being released.
+
+**Example**:
+
+- Build-time SBOM of the `ubi9-micro` container image along with all of its installed components (RPMs):
+  [build/ubi9-micro-container-9.4-6.1716471860_amd64](https://github.com/RedHatProductSecurity/security-data-guidelines/blob/main/sbom/examples/container_image/build/ubi9-micro-container-9.4-6.1716471860_amd64.spdx.json)
+- Release-time SBOM of the same `ubi9-micro` container image:
+  [release/ubi9-micro-container-9.4-6.1716471860_amd64](https://github.com/RedHatProductSecurity/security-data-guidelines/blob/main/sbom/examples/container_image/release/ubi9-micro-container-9.4-6.1716471860_amd64.spdx.json)
+
+The difference between these two SBOMs is relatively minor; the release-time SBOM contains information on where the
+container image (identified by two distinct purls) itself is available from:
+
+```diff
+--- sbom/examples/container_image/build/ubi9-micro-container-9.4-6.1716471860_amd64.spdx.json
++++ sbom/examples/container_image/release/ubi9-micro-container-9.4-6.1716471860_amd64.spdx.json
+@@ -22,12 +22,12 @@
+         {
+           "referenceCategory": "PACKAGE-MANAGER",
+           "referenceType": "purl",
+-          "referenceLocator": "pkg:oci/ubi-micro@sha256:1c8483e0fda0e990175eb9855a5f15e0910d2038dd397d9e2b357630f0321e6d?arch=amd64"
++          "referenceLocator": "pkg:oci/ubi-micro@sha256%3A1c8483e0fda0e990175eb9855a5f15e0910d2038dd397d9e2b357630f0321e6d?arch=amd64&repository_url=registry.access.redhat.com/ubi9/ubi-micro&tag=9.4-6.1716471860"
```

Comment:
Another interesting difference here is that `:` gets percent-encoded to `%3A`. Should we be doing that to start with, or we don't care?

Reply:
Yea, we should percent-encode as per spec: https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst#rules-for-each-purl-component